api/python/cell_census/src/cell_census/open.py

```diff
@@ -3,36 +3,45 @@
 from typing import Any, Dict, Optional
 
 import s3fs
-import tiledb
 import tiledbsoma as soma
 
-from .release_directory import CensusLocator, CensusVersionDescription, get_census_version_description
+from .release_directory import CensusLocator, get_census_version_description
 from .util import uri_join
 
-DEFAULT_TILEDB_CONFIGURATION = {
+DEFAULT_TILEDB_CONFIGURATION: Dict[str, Any] = {
     # https://docs.tiledb.com/main/how-to/configuration#configuration-parameters
     "py.init_buffer_bytes": 1 * 1024**3,
     "soma.init_buffer_bytes": 1 * 1024**3,
 }
 
 
-def _open_soma(description: CensusVersionDescription) -> soma.Collection:
-    """Private."""
-    locator = description["soma"]
-    tiledb_config: Dict[str, Any] = {**DEFAULT_TILEDB_CONFIGURATION}
+def _open_soma(locator: CensusLocator, context: Optional[soma.options.SOMATileDBContext] = None) -> soma.Collection:
+    """Private. Merge config defaults and return open census as a soma Collection/context."""
+    context = context or soma.options.SOMATileDBContext()
     s3_region = locator.get("s3_region")
-    if s3_region is not None:
-        tiledb_config["vfs.s3.region"] = s3_region
-    return soma.Collection(
-        uri=locator["uri"], context=soma.options.SOMATileDBContext(tiledb_ctx=tiledb.Ctx(tiledb_config))
-    )
-
 
-# TODO:  there are several changes pending for `open_soma`:
-# * add config or context hook when it is further defined, allowing config overrides.
-# * port to the factory `soma.open()` when it exists
-#
-def open_soma(*, census_version: Optional[str] = "latest", uri: Optional[str] = None) -> soma.Collection:
+    if not context:
+        # if no user-defined context, cell_census defaults take precedence over SOMA defaults
+        tiledb_config = DEFAULT_TILEDB_CONFIGURATION
+        if s3_region is not None:
+            tiledb_config.copy()["vfs.s3.region"] = s3_region
```

Comment:
Isn't this a no-op?

Reply:
which part?   The assignment of region is definitely not a noop (i.e., it sets the config to the region specified by the cell census directory), and leaves it unset (ie. use defaults) when the directory lacks a region.

That said, I'm often _non-pythonic_ so if I'm doing something silly here, I would love to learn!

Reply:
My understanding is that calling `.copy()` will return a copy of the dictionary, which you're modifying, but since you're not assigning to anything the result will be lost.

Reply:
oh my - good call. I'm clearly at the point I can't see forest for trees in my code!

Reply:
OK, I cleaned it up - whew, good catch.